src/api/spec/controllers/source_controller_spec.rb

```diff
@@ -50,4 +51,18 @@
     it { expect(response).to be_success }
     it { expect(project.config.to_s).to include('Updated', 'by', 'test') }
   end
+
+  describe "PUT #update_project_meta" do
+    it "raises exception for invalid DownloadRepository" do
+      login admin
+      bypass_rescue
+
+      xml = ActiveXML::Node.new(project.to_axml)
+      repo = xml.add_element 'repository', { name: "download_on_demand" }
+      repo.add_element 'download', { arch: "i586", url: "http://somewhere", repotype: "INVALID" }
```

Comment:
It would be nice to separate this xml generating stuff into a `before` or a `let`.